### PR TITLE
add removal and change api `remove` to `destroy`

### PR DESCRIPTION
**Pull request by kelvinabrokwa**
This change, as is, only works when render calls are not debounced. However, without debouncing, performance is worse.
I will continue to explore alternatives.

[EDIT]: This has nothing to do with debouncing
This should unblock tests.

### CODE REVIEW

**Comment by mcwhittemore** [comment]
@kelvinabrokwa - why do you want to change remove to destroy?

**Comment by kelvinabrokwa** [comment]
@mcwhittemore https://github.com/mapbox/mapbox-gl-draw/issues/153

**Comment by kelvinabrokwa** [comment]
@mcwhittemore :eyes: plz

File: src/api.js [hunk 1]
```diff
@@ -185,4 +185,10 @@ export default class API extends mapboxgl.Control {
     return this;
   }
 
+  remove() {
```
Comment (mcwhittemore) [inline src/api.js]: Lets move this to draw.js, as its being used by mapbox-gl

Comment (kelvinabrokwa) [inline src/api.js]: done

**Comment by mcwhittemore** [comment]
@kelvinabrokwa - I'm getting an error when trying to load debug.

image

**Comment by kelvinabrokwa** [comment]
@mcwhittemore yeah... so it turns out there is no ~public~ way to know if `load` has fired and since draw depends on that to set styles and sources it is better to `map.addControl` in the user's `map.on('load')` callback. I need to document this.

**Comment by mcwhittemore** [comment]
@kelvinabrokwa can you update debug to not fail and document what needs to be documented?

**Comment by kelvinabrokwa** [comment]
@mcwhittemore debug update as of fd1ffad and docs coming up!

**Comment by kelvinabrokwa** [comment]
@mcwhittemore https://github.com/mapbox/mapbox-gl-draw/blob/068235558bbc0eec781b50df47c97409ca7c5ce9/src/draw.js#L73 is a bit of a hack but it lets you safely add draw to the map before load fires. The upside is better backwards compatibility (and no need for new docs)

**Comment by mcwhittemore** [comment]
Merge, version, push and publish!

**Pull request merged**